### PR TITLE
Add context sections TUI windows

### PR DESCRIPTION
**Pull request by peace-maker**
This adds `pwndbg_[section]` tui windows to be used in a layout.
You can arrange `pwndbg_regs` or `pwndbg_disasm` as you wish.

The horizontal scrolling and truncation has to consider ANSI escape codes and makes sure to include all of them while scrolling through to properly keep the colors.

Since there is no event fired when TUI mode is enabled/disabled, we have to check `.is_valid()` whenever context data is available or the TUI is rendered to redirect context output appropriately again.

```
tui new-layout pwndbg {-horizontal { { -horizontal { pwndbg_code 2 pwndbg_disasm 8 } 2 { pwndbg_legend 1 pwndbg_regs 6 pwndbg_stack 6 } 3 } 7 cmd 3 } 3 { pwndbg_backtrace 1 } 1 } 1 status 1
layout pwndbg
```

grafik

### CODE REVIEW

File: FEATURES.md [hunk 1]
```diff
@@ -69,6 +69,14 @@ import splitmind
 end
 ```
 
+The context sections are available as native gdb TUI windows as well as `pwndbg_[sectionname]` windows.
+
+Try creating a layout and selecting it:
+```
+tui new-layout pwndbg {-horizontal { { -horizontal { pwndbg_code 2 pwndbg_disasm 8 } 2 { pwndbg_legend 1 pwndbg_regs 6 pwndbg_stack 6 } 3 } 7 cmd 3 } 3 { pwndbg_backtrace 1 } 1 } 1 status 1
+layout pwndbg
+```
```
Comment (disconnect3d) [inline FEATURES.md:78]: May be worth adding a screenshot here

File: pwndbg/dbg/gdb.py [hunk 2]
```diff
@@ -973,17 +972,15 @@ def setup(self):
...
         set confirm off
         set verbose off
         set pagination off
-        set height 0
         set history save on
         set follow-fork-mode child
         set backtrace past-main on
         set step-mode on
         set print pretty on
-        set width {pwndbg.ui.get_window_size()[1]}
```
Comment (disconnect3d) [inline pwndbg/dbg/gdb.py:986]: Are those changes required in this PR? Why we need them?

Comment (disconnect3d) [reply pwndbg/dbg/gdb.py:986]: Explanation: https://github.com/pwndbg/pwndbg/commit/30bfa8f1b8a45e83fa1f79281f53b8c9c07fb96d

TL;DR: GDB updates width/height automatically
